src/routes/(app)/strafbakken/+page.svelte

```diff
@@ -1,15 +1,131 @@
-<script>
+<script lang="ts">
+  import type { PageData } from "../$types";
+  import { Plus, Minus } from "svelte-heros-v2";
+  import Modal from "./Modal.svelte";
+  import { openModal } from "svelte-modals";
+  import type { User } from "@prisma/client";
 
+  import { Modals, closeModal } from "svelte-modals";
+
+  interface sb extends User {
+    _count: {
+      StrafbakReceived: number;
+    };
+  }
+
+  interface sbPageData extends PageData {
+    strafbakken: sb[];
+  }
+
+  export let data: sbPageData;
+
+  const trekBak = (id: number, index: number) => {
+    changeCount(index, -1);
+    fetch("/strafbakken", {
+      method: "DELETE",
+      body: JSON.stringify({
+        user: id,
+      }),
+    }).catch(console.error);
+  };
+
+  const changeCount = (index: number, n: number) => {
+    data.strafbakken[index]._count.StrafbakReceived += n;
+  };
 </script>
 
-<div>
-  <h1>
-    Success Naut
-  </h1>
-</div>
+<main>
+  <Modals>
+    <!-- svelte-ignore a11y-click-events-have-key-events -->
+    <div slot="backdrop" class="backdrop" on:click={closeModal} />
+  </Modals>
+  <table-container>
+    <table>
+      <thead>
+        <tr>
+          <th>Naam</th>
+          <th>Bakken</th>
+          <th>Acties</th>
+        </tr>
+      </thead>
+      <tbody>
+        {#each data.strafbakken as user, i}
+          <tr class="p-20">
+            <td>{user.nickname || user.firstName}</td>
+            <td>{user._count.StrafbakReceived}</td>
+            <td class="actions">
+              <Plus
+                class="cursor-pointer hover:invert-[.35] transition z-0 focus:outline-0"
+                on:click={() =>
+                  openModal(Modal, {
+                    username: user.nickname || user.firstName,
+                    uid: user.id,
+                    changeCount,
+                    index: i,
+                  })}
+              />
+              <Minus
+                class={user._count.StrafbakReceived
+                  ? "cursor-pointer hover:invert-[.35] transition z-0 focus:outline-0"
+                  : "invert-[.6] transition z-0 focus:outline-0"}
+                on:click={user._count.StrafbakReceived
+                  ? () => trekBak(user.id, i)
+                  : null}
+              />
+            </td>
+          </tr>
+        {/each}
+      </tbody>
+    </table>
+  </table-container>
+</main>
+
+<style lang="scss">
+  $tr-padding: 0.75rem;
+
+  table-container {
+    width: 100%;
+    place-items: center;
+    display: grid;
+
+    td,
+    th {
+      padding: $tr-padding;
+      text-align: left;
+    }
+
+    tbody {
+      tr {
+        transition: all 0.4s ease;
+
+        &:nth-child(odd) {
+          background-color: #d3c0ff;
+        }
+
+        // &:has(td:not(.actions):hover) {
```

Comment:
comments

Reply:
ja, je kan er uiteindelijk op klikken en daarvoor had ik de styles al maar omdat ik dat nu nog niet gemaakt heb, heb ik ook de 'clickable' styles even weggelaten in deze branch, en teruggezet in de strafbakken branch